classes/author/Repository.inc.php

```diff
@@ -0,0 +1,226 @@
+    {
+        $schemaService = Services::get('schema');
+
+        $validator = ValidatorFactory::make(
+            $props,
+            $schemaService->getValidationRules(PKPSchemaService::SCHEMA_AUTHOR, $allowedLocales)
+        );
+
+        // Check required fields
+        ValidatorFactory::required(
+            $validator,
+            $author,
+            $schemaService->getRequiredProps(PKPSchemaService::SCHEMA_AUTHOR),
+            $schemaService->getMultilingualProps(PKPSchemaService::SCHEMA_AUTHOR),
+            $allowedLocales,
+            $primaryLocale
+        );
+
+        // Check for input from disallowed locales
+        ValidatorFactory::allowedLocales($validator, $schemaService->getMultilingualProps(PKPSchemaService::SCHEMA_AUTHOR), $allowedLocales);
+
+        // The publicationId must match an existing publication that is not yet published
+        $validator->after(function ($validator) use ($props) {
+            if (isset($props['publicationId']) && !$validator->errors()->get('publicationId')) {
+                $publication = Repo::publication()->get($props['publicationId']);
+                if (!$publication) {
+                    $validator->errors()->add('publicationId', __('author.publicationNotFound'));
+                } elseif ($publication->getData('status') === PKPSubmission::STATUS_PUBLISHED) {
+                    $validator->errors()->add('publicationId', __('author.editPublishedDisabled'));
+                }
+            }
+        });
+
+        if ($validator->fails()) {
+            $errors = $schemaService->formatValidationErrors($validator->errors(), $schemaService->get(PKPSchemaService::SCHEMA_AUTHOR), $allowedLocales);
+        }
+
+        HookRegistry::call('Author::validate', [&$errors, $author, $props, $allowedLocales, $primaryLocale]);
+
+        return $errors;
+    }
+
+    /**
+     * @copydoc \PKP\services\entityProperties\EntityWriteInterface::add()
+     */
+    public function add(Author $author): int
+    {
+        $authorId = $this->dao->insert($author);
+        $author = Repo::author()->get($authorId);
+
+        HookRegistry::call('Author::add', [&$author]);
+
+        return $author->getId();
+    }
+
+    /**
+     * @copydoc \PKP\services\entityProperties\EntityWriteInterface::edit()
+     */
+    public function edit(Author $author, array $params)
+    {
+        $newAuthor = Repo::author()->newDataObject(array_merge($author->_data, $params));
+
+        HookRegistry::call('Author::edit', [&$newAuthor, $author, $params]);
+
+        $this->dao->update($newAuthor);
+
+        Repo::author()->get($newAuthor->getId());
+    }
+
+    /**
+     * @copydoc \PKP\services\entityProperties\EntityWriteInterface::delete()
+     */
+    public function delete(Author $author)
+    {
+        HookRegistry::call('Author::delete::before', [&$author]);
+        $this->dao->delete($author);
+        HookRegistry::call('Author::delete', [&$author]);
```

Comment:
The references (`&`) shouldn't be needed on this or any of the hooks above.

Reply:
Removed